docs/reference/api/storage/BufferedBlockDevice.md

```diff
@@ -0,0 +1,23 @@
+## BufferedBlockDevice
+
+<span class="images">![](https://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/class_slicing_block_device.png)<span>BufferedBlockDevice class hierarchy</span></span>
+
+The BufferedBlockDevice class is a block device adpator, whose purpose is to reduce the read and program sizes of the underlying block device to 1. Large read and program sizes may make life difficult for block device users, so BufferedBlockDevice reduces both sizes to the minimum, where reads and writes to the underlying block device use an internal buffer. Calling the `sync` API ensures writes are flushed to the underlying block device.
```

Comment:
adapter
e underlying block device to 1 byte.
I think we better use the word "write" instead of "program" because "program sizes" may be understood as program==code. I know the interface of write in BD is "to program" but still it will be more clear

Reply:
The term `program` is the one used by block devices, hence its usage.